### PR TITLE
Appending warn to changelog about breaking change

### PR DESCRIPTION
**Pull request by pedro-stanaka**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

I am just adding a warn to the change log as per suggestion of @fpetkovski and @MichaHoffmann.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 
 - [#7317](https://github.com/thanos-io/thanos/pull/7317) Tracing: allow specifying resource attributes for the OTLP configuration.
 - [#7367](https://github.com/thanos-io/thanos/pull/7367) Store Gateway: log request ID in request logs.
-- [#7361](https://github.com/thanos-io/thanos/pull/7361) Query: pass query stats from remote execution from server to client.
+- [#7361](https://github.com/thanos-io/thanos/pull/7361) Query: pass query stats from remote execution from server to client. Warning: we changed the protobuf of the QueryAPI, if you use `query.mode=distributed` you need to update your client (upper level Queriers) first, before updating leaf Queriers (servers).
```
Comment (MichaHoffmann) [inline CHANGELOG.md:24]: I think the usual thing we do would be to start with `Query: *breaking :warning:* ....`

**Comment by MichaHoffmann** [comment]
Thank you for the followup!

File: CHANGELOG.md (hunk 1 above)
Comment (pedro-stanaka) [reply CHANGELOG.md]: Prepended that to the message. Thx for the review.

**Review by MichaHoffmann — APPROVED** [review]
LGTM

**Pull request merged**